modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/test/StartService.java

```diff
@@ -159,39 +164,162 @@ private void clearPendingConnectors() {
 
     private String getServiceURL(ServiceInfo service) {
         try {
-            String listenerInterface = Constants.DEFAULT_INTERFACE;
-            String basePath = service.getName();
-
-            AnnAttachmentInfo annotation = service.getAnnotationAttachmentInfo(Constants.HTTP_PACKAGE_PATH,
-                    Constants.ANN_NAME_CONFIG);
-            if (annotation != null) {
-                basePath = annotation.getAttributeValue(Constants.ANN_CONFIG_ATTR_BASE_PATH)
-                        .getStringValue();
-            }
-            if (basePath.startsWith("\"")) {
-                basePath = basePath.substring(1, basePath.length() - 1);
+            AnnAttachmentInfo annotationInfo = service.getAnnotationAttachmentInfo(Constants
+                    .HTTP_PACKAGE_PATH, Constants.ANN_NAME_CONFIG);
+
+            String basePath = discoverBasePathFrom(service, annotationInfo);
+            Set<ListenerConfiguration> listenerConfigurationSet = getDefaultOrDynamicListenerConfig(annotationInfo);
+
+            if (listenerConfigurationSet.isEmpty()) {
+                throw new BallerinaException(MSG_PREFIX + "Cannot find relevant transport listener configuration");
             }
 
-            if (!basePath.startsWith("/")) {
-                basePath = "/".concat(basePath);
+            //There can be multiple urls with the new transport implementation, so for the time being,
+            // just use the first listener configuration to generate the url
+            ListenerConfiguration config = listenerConfigurationSet.iterator().next();
+            String host = config.getHost().equals(DEFAULT_HOSTNAME) ? LOCALHOST : config.getHost();
+            int port = config.getPort();
+            String scheme = config.getScheme();
+            return new URL(scheme, host, port, basePath).toExternalForm();
+        } catch (MalformedURLException e) {
+            throw new BallerinaException(MSG_PREFIX + "Error while constructing service url for " + service.getName());
+        }
+
+    }
+
+    private Set<ListenerConfiguration> getDefaultOrDynamicListenerConfig(AnnAttachmentInfo annotationInfo) {
```

Comment:
these were copied method from ballerina for the time being(since these are private method in ballerina), need to move them to a util method in ballerina side and use them here, anyway will change the method as you said here